client/src/api.ts

```diff
@@ -1,5 +1,108 @@
-// import axios from "axios";
+import axios from "axios";
 
-const DIALOG_ENDPOINT = process.env.DIALOG_ENDPOINT || "/dialog/";
+export interface SuccessfulCreationResponse {
+  data: {
+    status: number;
+    lessonID: string;
+    sessionInfo: {
+      sessionId: string;
+      sessionHistory: string;
+      previousUserResponse: string;
+      previousSystemResponse: string[];
+      dialogState: {
+        expectationsCompleted: boolean;
+        expectationData: {
+          ideal: string;
+          score: number;
+          satisfied: boolean;
+          status: string;
+        }[];
+        hints: boolean;
+      };
+      hash: string;
+    };
+    response: { author: string; type: string; data: { text: string } }[];
+  };
+  status: number;
+}
 
+export interface SuccessfulContinuationResponse {
+  data: {
+    status: number;
+    lessonID: string;
+    sessionInfo: {
+      sessionId: string;
+      sessionHistory: string;
+      previousUserResponse: string;
+      previousSystemResponse: string[];
+      dialogState: {
+        expectationsCompleted: boolean;
+        expectationData: {
+          ideal: string;
+          score: number;
+          satisfied: boolean;
+          status: string;
+        }[];
+        hints: boolean;
+      };
+      hash: string;
+    };
+    response: { author: string; type: string; data: { text: string } }[];
+    sentToGrader: boolean;
+    completed: boolean;
+    score: number;
+    expectationActive: number;
+  };
+  status: number;
+}
+
+export interface FailedResponse {
+  status: number;
+  statusText: string;
+  data: string;
+}
+
+const DIALOG_ENDPOINT = process.env.DIALOG_ENDPOINT || "/dialog";
+//{status:number; lessonID:string, sessionInfo: {sessionId:string, sessionHistory:string, previousUserResponse:string, previousSystemResponse:string[],dialogState:{expectationsCompleted:boolean, expectationData:{ideal:string, score:number, satisfied:boolean, status:string}[],hints:boolean},hash:string}, response:{author:string, type:string, data:{text:string}}[]};
 // put dialog api calls here
+export async function createSession(
+  lesson: string
+): Promise<SuccessfulCreationResponse | FailedResponse> {
+  try {
+    return await axios.post(`${DIALOG_ENDPOINT}/${lesson}`, {});
+  } catch (error) {
+    console.log(error.response);
+    return error.response;
+  }
+}
+
+export async function continueSession(props: {
+  lesson: string;
+  session: {
```

Comment:
two things:

- `DialogRequest` should be it's own interface (better than inlined into the function params)
- import and use the existing interfaces where they apply I think here it's `SessionData`

So it can be simplified to something like

```typescript
export interface DialogRequest {
  lesson: string;
  session: SessionData;
  outboundChat: string
}
```